hera_cal/nucal.py

```diff
@@ -15,6 +15,20 @@
 from jax import numpy as jnp
 jax.config.update("jax_enable_x64", True)
 
+# Optional import of Optax library for optimization
+try:
+    import optax
+
+    # Approved Optax Optimizers
+    OPTIMIZERS = {
+        'adabelief': optax.adabelief, 'adafactor': optax.adafactor, 'adagrad': optax.adagrad, 'adam': optax.adam,
+        'adamw': optax.adamw, 'fromage': optax.fromage, 'lamb': optax.lamb, 'lars': optax.lars,
+        'noisy_sgd': optax.noisy_sgd, 'dpsgd': optax.dpsgd, 'radam': optax.radam, 'rmsprop': optax.rmsprop,
+        'sgd': optax.sgd, 'sm3': optax.sm3, 'yogi': optax.yogi
+    }
+except:
+    warnings.warn('Optax is not installed. Some functionality may not be available')
```

Comment:
Seems like you're adding optax to setup.py, so why are you saying it's optional?

Reply:
Changed this so that it's no longer optional